net/dialer.go

```diff
@@ -46,9 +40,15 @@ func (s *server) dial(peerID libpeer.ID) (pb.ServiceClient, error) {
 			return pb.NewServiceClient(conn), nil
 		}
 	}
-	ctx, cancel := context.WithTimeout(context.Background(), DialTimeout)
-	defer cancel()
-	conn, err := grpc.DialContext(ctx, peerID.String(), s.opts...)
+	// Need "passthrough" since we moved from [grpc.DialContext] to [grpc.NewClient].
```

Comment:
todo: Can you please expand a little as to what this means, the linked docs are far longer than I would like to read in order to either review or skim over this when maintaining.

Reply:
suggestion:

> since we moved from [grpc.DialContext] to [grpc.NewClient]

Given that there is no todo attached to the comment, this seems irrelevant to understanding the current (new) code.  If we need a comment here, I suggest that describing the code history is not part of that comment.

Reply:
sure will do.

Reply:
- [x] document that without passthrough it will assume dns target by default